apps/system/test/unit/nfc_handover_manager_test.js

```diff
@@ -217,56 +476,99 @@ suite('Nfc Handover Manager Functions', function() {
     var cps;
     var mac;
 
-    var spyPairing;
-    var spySendNDEF;
-
     setup(function() {
+      this.sinon.useFakeTimers();
       cps = NDEF.CPS_ACTIVE;
       mac = '01:23:45:67:89:AB';
-      spyPairing = this.sinon.spy(nfcHandoverManager, '_doPairing');
-      spySendNDEF = this.sinon.spy(MockMozNfc.MockNFCPeer, 'sendNDEF');
-
+      this.sinon.spy(nfcHandoverManager, '_doAction');
+      this.sinon.spy(nfcHandoverManager, '_saveBluetoothStatus');
       nfcHandoverManager.start();
-      invokeBluetoothGetDefaultAdapter();
     });
 
     teardown(function() {
-      spyPairing.restore();
-      spySendNDEF.restore();
       nfcHandoverManager.stop();
+      nfcHandoverManager._adapter = null;
+    });
+
+    test('_handleHandoverRequest(): aborting via empty Hs', function() {
+      this.sinon.stub(Service, 'query')
+        .withArgs('BluetoothTransfer.isFileTransferInProgress').returns(true);
+      this.sinon.spy(MockMozNfc.MockNFCPeer, 'sendNDEF');
+      var handoverRequest = NDEFUtils.encodeHandoverRequest(mac, cps);
+      var hs = NDEFUtils.encodeEmptyHandoverSelect();
+      nfcHandoverManager._handleHandoverRequest(handoverRequest,
+        MockMozNfc.MockNFCPeer);
+
+      assert.isTrue(MockMozNfc.MockNFCPeer.sendNDEF.calledWith(hs));
     });
 
     test('_handleHandoverRequest(): sends Hs message to peer', function() {
```

Comment:
I think a better description would be, 'proper handling when file transfer is not in progress'. We're not exactly sending the Hs here.